src/uaserver.cpp

```diff
@@ -35,50 +36,6 @@
 #include <boost/lexical_cast.hpp>
 #endif // HAS_SERVERCONFIG_LOADER
 
-static Log::LogComponentHandle logItComponentHandle = Log::INVALID_HANDLE;
-
-static void logFromOpen62541 (
-		void *logContext,
-		UA_LogLevel level,
-		UA_LogCategory category,
-        const char *msg,
-		va_list args)
-{
-	// translate open62541 log level into LogIt log level
-	Log::LOG_LEVEL logItLogLevel;
-	switch (level)
-	{
-		case UA_LOGLEVEL_DEBUG: logItLogLevel = Log::DBG; break;
-		case UA_LOGLEVEL_ERROR: logItLogLevel = Log::ERR; break;
-		case UA_LOGLEVEL_FATAL: logItLogLevel = Log::ERR; break;
-		case UA_LOGLEVEL_INFO: logItLogLevel = Log::INF; break;
-		case UA_LOGLEVEL_TRACE: logItLogLevel = Log::TRC; break;
-		case UA_LOGLEVEL_WARNING: logItLogLevel = Log::WRN; break;
-		default: logItLogLevel = Log::ERR; // just in case they added a new level.
-	}
-	// translate open62541 category.
-	std::string categoryStr ("category_unknown");
-	switch (category)
-	{
-		case UA_LOGCATEGORY_CLIENT: categoryStr = "client"; break;
-		case UA_LOGCATEGORY_NETWORK: categoryStr = "network"; break;
-		case UA_LOGCATEGORY_SECURECHANNEL: categoryStr = "securechannel"; break;
-		case UA_LOGCATEGORY_SECURITYPOLICY: categoryStr = "securitypolicy"; break;
-		case UA_LOGCATEGORY_SERVER: categoryStr = "server"; break;
-		case UA_LOGCATEGORY_SESSION: categoryStr = "session"; break;
-		case UA_LOGCATEGORY_USERLAND: categoryStr = "userland"; break;
-	}
-	char line [1024] = {0};
-	vsnprintf(line, sizeof line-1, msg, args);
```

Comment:
Might be useful to add a check to the return value based on the definition below
```
The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is non-negative and less than n, the string has been completely written.
```

Reply:
And, if not the whole msg was produced, do what?

Reply:
- Easiest
Do nothing accepting that the possibility is low and the impact potentially insignificant

- Easy
Mention in the LOG below that the message is truncated

- More tedious
Use the returned value to allocate dynamically the needed space for all the characters and pass that.
In case of negative value, log that there was an encoding error (or that is what is returned in Windows in any case)